compose/config/validation.py

```diff
@@ -162,10 +163,25 @@ def _parse_oneof_validator(error):
         Inspecting the context value of a ValidationError gives us information about
         which sub schema failed and which kind of error it is.
         """
+
+        required = [context for context in error.context if context.validator == 'required']
+        if required:
+            return required[0].message
+
+        additionalProperties = [context for context in error.context if context.validator == 'additionalProperties']
+        if additionalProperties:
+            invalid_config_key = _parse_key_from_error_msg(additionalProperties[0])
+            return "contains unsupported option: '{}'".format(invalid_config_key)
+
         constraint = [context for context in error.context if len(context.path) > 0]
         if constraint:
             valid_types = _parse_valid_types_from_validator(constraint[0].validator_value)
-            msg = "contains {}, which is an invalid type, it should be {}".format(
+            path = [
```

Comment:
This feels a bit like overkill to me. The contents of the `path` deque will be string keys, so I don't think we need to test for the instance of string here.

The depth of config keys is only two deep. So we already have the service name, already have the config key before this function is called, so we can confidently at this point look at just `constraint[0].path`.

`        if constraint:
            valid_types = _parse_valid_types_from_validator(constraint[0].validator_value)
            msg = "'{}' contains {}, which is an invalid type, it should be {}".format(
                "".join(constraint[0].path),
                constraint[0].instance,
                valid_types
            )
`